IdentityCore/src/webview/MSIDWebviewAuthorization.m

```diff
@@ -26,33 +26,213 @@
+                   context:context
+         completionHandler:completionHandler];
+}
 
-+ (id<MSIDWebviewInteracting>)embeddedWebviewControllerWithRequestParameters:(MSIDRequestParameters *)parameters
-                                                                     factory:(MSIDOAuth2Factory *)factory;
++ (void)startEmbeddedWebviewWebviewAuthWithRequestParameters:(MSIDRequestParameters *)parameters
+                                                     webview:(WKWebView *)webview
+                                                     factory:(MSIDOauth2Factory *)factory
+                                                     context:(id<MSIDRequestContext>)context
+                                           completionHandler:(MSIDWebUICompletionHandler)completionHandler
 {
-    return nil;
+    id<MSIDWebviewInteracting> embeddedWebviewController = [factory embeddedWebviewControllerWithRequest:parameters
+                                                                                           customWebview:webview
+                                                                                                 context:context
+                                                                                       completionHandler:completionHandler];
+    [self startWebviewAuth:embeddedWebviewController
+                   context:context
+         completionHandler:completionHandler];
 }
 
-+ (id<MSIDWebviewInteracting>)embeddedWebviewControllerWithRequestParameters:(MSIDRequestParameters *)parameters
-                                                                     webview:(WKWebView *)webview
-                                                                     factory:(MSIDOAuth2Factory *)factory;
++ (void)startSystemWebviewWebviewAuthWithRequestParameters:(MSIDRequestParameters *)parameters
+                                                   factory:(MSIDOauth2Factory *)factory
+                                                   context:(id<MSIDRequestContext>)context
+                                         completionHandler:(MSIDWebUICompletionHandler)completionHandler
 {
+
+    id<MSIDWebviewInteracting> systemWebviewController = [factory systemWebviewControllerWithRequest:parameters
+                                                                                   callbackURLScheme:parameters.redirectUri
+                                                                                             context:context
+                                                                                   completionHandler:completionHandler];
+    [self startWebviewAuth:systemWebviewController
+                   context:context
+         completionHandler:completionHandler];
+}
+
++ (void)startWebviewAuth:(id<MSIDWebviewInteracting>)webviewController
+                 context:(id<MSIDRequestContext>)context
+       completionHandler:(MSIDWebUICompletionHandler)completionHandler
+{
+    if (![self setCurrentWebSession:webviewController])
```

Comment:
seems we are not able to `setCurrentWebSession` for the second time, as there is no code to release `s_currentWebSession` when a webview call is completed? :)

Reply:
I've included the session cancelling as discussed before